src/components/Popover.js

```diff
@@ -1,110 +1,84 @@
-import React from 'react';
+import React, { useState, useRef } from 'react';
 import PropTypes from 'prop-types';
 import MuiPopover from '@material-ui/core/Popover';
 import IconButton from '@material-ui/core/IconButton';
-import { findDOMNode } from 'react-dom';
 import CloseIcon from '@material-ui/icons/Close';
 
-class Popover extends React.Component {
-  state = {
-    open: false,
-  };
+const Popover = ({ className, trigger, refExit, content, ...providedProps }) => {
+  const [isOpen, open] = useState(false);
+  const anchorEl = useRef(null);
 
-  UNSAFE_componentWillMount() {
-    this.anchorEl = null;
-  }
+  const handleClick = event => {
+    anchorEl.current = event.currentTarget;
+    open(true);
+  };
 
-  componentDidMount() {
-    if (this.props.refClose) {
-      this.props.refClose(this.handleRequestClose);
+  const handleRequestClose = callback => {
+    open(false);
+    if (callback && typeof callback === 'function') {
+      callback();
     }
-  }
+  };
 
-  componentDidUpdate(prevProps, prevState) {
-    /*
-     * Update Popover position if a filter removes data from the table because
-     * it affects the window height which would cause the Popover to in the wrong place
-     */
-    if (this.state.open === true) {
-      this.anchorEl = findDOMNode(this.anchorEl);
-      if (this.popoverActions) {
-        this.popoverActions.updatePosition();
-      }
-      const shouldHide = typeof this.props.hide === 'boolean' ? this.props.hide : false;
-      if (shouldHide) {
-        this.handleRequestClose();
-      }
-    }
+  let closeIconClass;
```

Comment:
It seems unnecessary, but I'll keep it for now.
Because, I do not know the exact reason.